### DISCUSSION TITLE
Variants of sof-hda-generic and why some options are different ( DYNAMIC / DEEP_BUFFER )

### DISCUSSION THREAD

**Discussion by smunaut** [thesofproject/sof#7953] [Q&A]
Hi,

So I'm looking specifically at `sof-hda-generic-2ch` here.
The one provided by my distribution in the package comes from `topology/topology1/kernel_dependent/v5.19/`.
Now there is another variant in `topology/topology1/` and that one is almost exacty the same except for the `DEEP_BUFFER` is not enabled.

Then I'm also looking at the `topology/topology1/dsp_enhancement/sof-hda-generic-eq-2ch.tplg` and that one is also built without `DEEP_BUFFER` but also without the `DYNAMIC` options.  (and obviously has the EQ filters added, but that I understand)

So my questions is: What are those options and why are they different between those variants ? Like my understanding of `DEEP_BUFFER` is is that it adds another DMA path input with a big buffer, but not why some have it and some don't. And I have no idea what `DYNAMIC` means and why the EQ variant wouldn't have it.

Cheers,

Sylvain

**Comment by plbossart** [comment]
"kernel_dependent" means exactly that, we cannot release topologies that contain features that are kernel-dependent or we would incur the risk of breaking user setups.

the 'DYNAMIC' part refers to dynamic pipelines, where pipelines are only instantiated when used, instead of being created at DSP boot
'Deep buffer' refers to a larger buffer in the DSP with a mixer, that also has kernel dependencies.

**Reply by smunaut** [reply]
Thanks.

* But the topologies in the sof-bin (2.2.6) package are the one from "kernel_dependent" which is what confused me. So I guess it assumes a kernel >= 5.19.
* What's the advantage of DYNAMIC and DEEP_BUFFER ?
  * I guess DYNAMIC might be a resource usage thing where you can have more pipelined defined that would be able to run at once.
  * And Deep Buffer would be for cases where the CPU is heavily loaded and can't refill a short buffer in time.
